.github/workflows/tests-pro-integration.yml

```
@@ -169,11 +181,15 @@ jobs:
         if: inputs.disableCaching != true
         uses: actions/cache@v3
         with:
+          # TODO: maybe we should split these into two steps with different keys
           path: |
             localstack-ext/.venv
             localstack/.filesystem/var/lib/localstack
           # include the matrix group (to re-use the var-libs used in the specific test group)
-          key: community-it-${{ runner.os }}-python-${{ steps.setup-python.outputs.python-version }}-venv-${{ hashFiles('localstack-ext/setup.cfg', 'localstack-ext/pyproject.toml', 'localstack/setup.cfg', '**/packages.py', '**/packages/*') }}-${{steps.determine-companion-ref.outputs.result}}-group-${{ matrix.group }}
+          key: community-it-${{ runner.os }}-python-${{ steps.setup-python.outputs.python-version }}-venv-${{ hashFiles('localstack-ext/setup.cfg', 'localstack-ext/pyproject.toml','localstack-ext/**/packages.py','localstack-ext/packages/*.py', 'localstack/setup.cfg', 'localstack/**/packages.py',  'localstack/packages/*.py') }}-${{steps.determine-companion-ref.outputs.result}}-group-${{ matrix.group }}
+          restore-keys: |
+            community-it-${{ runner.os }}-python-${{ steps.setup-python.outputs.python-version }}-venv-${{ hashFiles('localstack-ext/setup.cfg', 'localstack-ext/pyproject.toml','localstack-ext/**/packages.py','localstack-ext/packages/*.py', 'localstack/setup.cfg', 'localstack/**/packages.py',  'localstack/packages/*.py') }}-master-
```

I think we should either implement the TODO you added (split the caches, one for the venv - used by both groups, and one for the package installers - used for each group invidually), or we have to add the group to the restore key as well.
Otherwise they would overwrite each other.
Also, I think the `${{steps.determine-companion-ref.outputs.result}}` would evaluate to `refs/heads/master` (and not `master`).

So I think the `restore-keys` should be changed to:

```suggestion
          restore-keys: |
            community-it-${{ runner.os }}-python-${{ steps.setup-python.outputs.python-version }}-venv-${{ hashFiles('localstack-ext/setup.cfg', 'localstack-ext/pyproject.toml','localstack-ext/**/packages.py','localstack-ext/packages/*.py', 'localstack/setup.cfg', 'localstack/**/packages.py',  'localstack/packages/*.py') }}-refs/heads/master-group-${{ matrix.group }}
```